pkg/controller/migmigration/task.go

```diff
@@ -184,11 +202,12 @@ func (t *Task) Run() error {
 	t.Log.Info("[RUN]", "stage", t.stage(), "phase", t.Phase)
 
 	t.init()
+	step := t.setStep()
```

Comment:
Please explain how this change is related to and required for this feature.  Else, revert.

Reply:
This is simple. That's the first bullet in the PR description - we need to cancel immediately, without allowing a stage to finish. In case it is waiting for backup, it will block for inadequate amount of time, and only then start the cancel process.

Reply:
@jortel In other words, this is the implementation for Itinerary swaps, which happens on Failure or Cancel.

Reply:
The bullet explains the desired behavior but not the need for the infrastructure change. 
Currently:
Each phase implementation is designed to run quickly.  Nothing waits.
At the beginning of each reconcile, Task.Run(), Task.init() determines the appropriate itinerary. 
The `Spec.Cancel` will trigger a reconcile and _new_ will immediately begin processing.
This code in `next()`:
```
	if current == -1 {
		if t.canceled() {
			t.Phase = CancelItinerary[0].phase
		} else {
			t.Phase = Completed
		}
		return
	}
```
Probably needs to get moved to `init()`.  I'm thinking we may need to store the itinerary in the Status or in an annotation so that `init()` can detect when the itinerary has changed and reset the Phase to the first phase.

Reply:
The itinerary will need to change as:
```
type Itinerary struct {
	Name string
	Steps []Step
}

var StageItinerary = Itinerary{
	Name: "Stage",
	Steps: []Step{
		{phase: Created},
		{phase: Started},
		{phase: Prepare},
		{phase: EnsureCloudSecretPropagated},
		{phase: AnnotateResources, all: HasPVs},
		{phase: EnsureStagePods, all: HasPVs},
		{phase: StagePodsCreated, all: HasStagePods},
		{phase: RestartRestic, all: HasStagePods},
		{phase: ResticRestarted, all: HasStagePods},
		{phase: QuiesceApplications, all: Quiesce},
		{phase: EnsureQuiesced, all: Quiesce},
		{phase: EnsureStageBackup, all: HasPVs},
		{phase: StageBackupCreated, all: HasPVs},
		{phase: EnsureStageBackupReplicated, all: HasPVs},
		{phase: EnsureStageRestore, all: HasPVs},
		{phase: StageRestoreCreated, all: HasPVs},
		{phase: EnsureStagePodsDeleted, all: HasStagePods},
		{phase: EnsureStagePodsTerminated, all: HasStagePods},
		{phase: EnsureAnnotationsDeleted, all: HasPVs},
		{phase: Completed},
	},
}
...
```
So that it can be named.